### PR TITLE
Show images inline where possible

### PR DESCRIPTION
**Pull request by Nerixyz**
This PR changes the Typst preview to be inline - next to the code - like it's done on the Typst docs.

To do this, the typst-preview remark plugin now transforms examples into MDX components, which is a bit more complicated, because it needs to emit an AST with the parsed MDX component (and parsed JS inside its arguments). Images are resolved "inside" the MDX file (resolving them in the component didn't really work, but I might've missed something). To have better editor integration, I've used TypeScript.

<sub>As a bonus, this gets rid of the errors about `<pre>` inside `<p>`.</sub>

I'd consider this a draft, as there are some things left to add/check, but **feel free to give an early review**.

- [x] Check all markdown files for large code blocks and add `big` if needed
- [ ] Add some more comments to the transformer (as to why things are done this way)
- [x] Check the Typst template (mentioned in https://github.com/cetz-package/docs/pull/4#issuecomment-2131362298)

<details><summary>Preview</summary>

firefox_2024-05-26_14-30-25

</details>

### CODE REVIEW

File: src/remark/typst.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+          attributes: [
+            {
+              type: "mdxJsxAttribute",
+              name: "imgUrl",
+              value: assetRequireAttributeValue(requireString),
+            },
+            {
+              type: "mdxJsxAttribute",
+              name: "isBig",
+              value: boolToJsx(node.meta?.includes("big")),
```
Comment (fenjalien) [inline src/remark/typst.ts:98]: Can we use the term "stack" or "vertical" instead of "big". We use the same term in the pdf manual for the examples

File: src/remark/typst.ts [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    });
+    await Promise.all(children);
+  };
+  return transformer;
+};
+
+export default plugin;
+
+// from https://github.com/facebook/docusaurus/blob/aab332c2ae3f78df5cdac6344bab0075ea07a035/packages/docusaurus-mdx-loader/src/remark/utils/index.ts#L103-L156
+function assetRequireAttributeValue(
```
Comment (fenjalien) [inline src/remark/typst.ts:113]: Oh my, please find a better way to do this

File: src/components/Type.jsx [hunk 3]
```diff
@@ -36,6 +36,6 @@ const types = {
 
 export default function Type({ children }) {
     const type = types[children]
-    return <span class={`type type-${type.class ?? "obj"}`}><a href={type.link}>{children}</a></span>
+    return <span className={`type type-${type.class ?? "obj"}`}><a href={type.link}>{children}</a></span>
```
Comment (fenjalien) [inline src/components/Type.jsx:39]: Why is this necessary?

File: src/remark/typst.ts (hunk 1 above)
Comment (Nerixyz) [reply src/remark/typst.ts]: It's "vertical" now to be the same as in the pdf manual.

File: src/remark/typst.ts (hunk 2 above)
Comment (Nerixyz) [reply src/remark/typst.ts]: I put the image as a child, which works because afterward a docusaurus plugin resolves the image and turns it into a MDX element.

File: src/components/Type.jsx (hunk 3 above)
Comment (Nerixyz) [reply src/components/Type.jsx]: I made that change to get rid of an error (you can't use `class` in React), but I removed the change now.

Comment (fenjalien) [reply src/components/Type.jsx]: Huh, I am not aware of that error. Please re-add it if you like.

**Comment by Nerixyz** [comment]
I changed the template to include `width: auto` and the CSS from flexbox to grid. I'm not a CSS wizard, so any improvements are welcome.

File: src/components/Type.jsx (hunk 3 above)
Comment (Nerixyz) [reply src/components/Type.jsx:39]: The `Parameter` component now has some warnings about `class` and a missing `key` prop. I could fix them here, but it might look weird in the final diff because this PR will be squashed.

**Review by fenjalien — CHANGES_REQUESTED** [review]
Looking good! Having some comments of whats going on would be super useful.

image
Could the images be on the left of the code block? Could you also make sure the code starts at the top of the block and not centered.
